### PR TITLE
dont invalidate cache on lns timeout, only on explicit negative

### PR DESCRIPTION
**Pull request by majestrate**

### CODE REVIEW

File: llarp/service/session.cpp [hunk 1]
```diff
@@ -27,7 +27,7 @@ namespace llarp
       const auto lastUsed = std::max(lastSend, lastRecv);
       if (lastUsed == 0s)
         return intro.IsExpired(now);
-      return now > lastUsed && (now - lastUsed > lifetime || intro.IsExpired(now));
+      return now > lastUsed && now - lastUsed > lifetime;
```
Comment (tewinget) [inline llarp/service/session.cpp]: probably keep the parentheses around the latter condition, just for ease of reading.

Comment (majestrate) [reply llarp/service/session.cpp]: fair

**Comment by majestrate** [comment]
rebased atop dev

**Review by tewinget — APPROVED** [review]
makes sense

**Pull request merged**